CRM/Contact/BAO/Contact/Permission.php

```diff
@@ -335,6 +439,104 @@ public static function relationship($selectedContactID, $contactID = NULL) {
 
 
   /**
+   * Filter a list of contact_ids by the ones that the
+   *  currently active user as a permissioned relationship with
+   *
+   * @param array $contact_ids
+   *   List of contact IDs to be filtered
+   *
+   * @return array
+   *   List of contact IDs that the user has permissions for
+   */
+  public static function relationshipList($contact_ids) {
+    $result_set = array();
+
+    // no processing empty lists (avoid SQL errors as well)
+    if (empty($contact_ids)) {
+      return array();
+    }
+
+    // get the currently logged in user
+    $config    = CRM_Core_Config::singleton();
```

Comment:
If you are only using this for one purpose you can also do
$useSecondDegreeUserPermissions = CRM_Core_Config::singleton()->secondDegRelPermissions